src/libcrun/linux.c

```diff
@@ -2736,29 +2736,116 @@ libcrun_set_oom (libcrun_container_t *container, libcrun_error_t *err)
   return 0;
 }
 
+const char *sysctlRequiringIPC[] = {
+  "kernel/msgmax",
+  "kernel/msgmnb",
+  "kernel/msgmni",
+  "kernel/sem",
+  "kernel/shmall",
+  "kernel/shmmax",
+  "kernel/shmmni",
+  "kernel/shm_rmid_forced",
+  NULL
+};
+
+static int
+validate_sysctl (const char *original_value, const char *name, unsigned long namespaces_created, libcrun_error_t *err)
+{
+  const char *namespace = "";
+
+  if (has_prefix (name, "fs/queue/"))
+    {
+      if (namespaces_created & CLONE_NEWIPC)
+        return 0;
+
+      namespace = "IPC";
+      goto fail;
+    }
+
+  if (has_prefix (name, "kernel/"))
+    {
+      size_t i;
+
+      for (i = 0; sysctlRequiringIPC[i]; i++)
+        if (strcmp (sysctlRequiringIPC[i], name) == 0)
+          {
+            if (namespaces_created & CLONE_NEWIPC)
+              return 0;
+
+            namespace = "IPC";
+            goto fail;
+          }
+
+      if (strcmp (name, "kernel.domainname") == 0)
```

Comment:
Should this be `"kernel/domainname"`?

Reply:
Yes I think in `runc` separator is replaced by `.` https://github.com/opencontainers/runc/blob/master/libcontainer/configs/validate/validator.go#L140